pkg/util/informermanager/interface.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package informermanager
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+)
+
+type SingleClusterInformerManager interface {
+	// Starts an informer for the given GroupResourceVersion if there isn't already one running. The new/existing
+	// informer is guaranteed to run as long as the given context remains uncancelled.
```

Comment:
What if `ForResource` is called multiple times with ctx of different deadlines?

Reply:
the underlying informer will only be stopped when the last context is done